ethcontract-common/src/artifact/hardhat.rs

```diff
@@ -0,0 +1,290 @@
+
+        Ok(artifact)
+    }
+
+    fn fill_artifact(
+        &self,
+        artifact: &mut Artifact,
+        export: HardHatExport,
+    ) -> Result<(), ArtifactError> {
+        if self.allowed(&export.chain_id) && self.allowed(&export.chain_name) {
+            for (name, contract_with_address) in export.contracts {
+                let ContractWithAddress {
+                    address,
+                    mut contract,
+                } = contract_with_address;
+
+                contract.name = name;
+
+                let mut contract = match artifact.get_mut(&contract.name) {
+                    Some(existing_contract) => {
+                        if existing_contract.abi != contract.abi {
+                            return Err(ArtifactError::AbiMismatch(contract.name));
+                        }
+
+                        existing_contract
+                    }
+                    None => artifact.insert(contract).inserted_contract,
+                };
+
+                contract.networks_mut().insert(
+                    export.chain_id.clone(),
+                    Network {
+                        address,
+                        deployment_information: None,
+                    },
+                );
+            }
+        }
+
+        Ok(())
+    }
+
+    fn fill_artifact_multi(
+        &self,
+        artifact: &mut Artifact,
+        export: HashMap<String, HashMap<String, HardHatExport>>,
+    ) -> Result<(), ArtifactError> {
+        for (chain_id, export) in export {
+            if !self.allowed(&chain_id) {
+                continue;
+            }
+
+            let mut multiple_nets = false;
+
+            for (chain_name, export) in export {
+                if !self.allowed(&chain_name) {
+                    continue;
+                }
+
+                if multiple_nets {
+                    return Err(ArtifactError::DuplicateChain(chain_id));
+                }
+
+                self.fill_artifact(artifact, export)?;
+
+                multiple_nets = true;
+            }
+        }
+
+        Ok(())
+    }
+
+    fn allowed(&self, name: &str) -> bool {
+        self.explicitly_allowed(name) && !self.explicitly_denied(name)
+    }
+
+    fn explicitly_allowed(&self, name: &str) -> bool {
+        self.networks_allow_list.is_empty() || self.networks_allow_list.iter().any(|x| x == name)
+    }
+
+    fn explicitly_denied(&self, name: &str) -> bool {
+        self.networks_deny_list.iter().any(|x| x == name)
+    }
+}
+
+impl Default for HardHatLoader {
+    fn default() -> Self {
+        HardHatLoader::new()
+    }
+}
+
+#[derive(Clone, Debug, Deserialize)]
+struct HardHatExport {
+    #[serde(rename = "name")]
+    chain_name: String,
+    #[serde(rename = "chainId")]
+    chain_id: String,
+
+    contracts: HashMap<String, ContractWithAddress>,
+}
+
+#[derive(Clone, Debug, Deserialize)]
+struct ContractWithAddress {
+    address: Address,
+    #[serde(flatten)]
+    contract: Contract,
+}
```

Comment:
Could you add some unit tests that maybe `load_from_json` with some `json!` value? It would be nice as it would also document the format a little in code.